app/code/Magento/Quote/README.md

```diff
@@ -1,11 +1,310 @@
+    - `order` is an order object (`\Magento\Sales\Model\Order` class)
+    - `quote` is a quote object (`\Magento\Quote\Model\Quote` class)
+
+- `sales_model_service_quote_submit_failure` event in the `\Magento\Quote\Model\QuoteManagement::rollbackAddresses` method. Parameters:
+    - `order` is an order object (`\Magento\Sales\Model\Order` class)
+    - `quote` is a quote object (`\Magento\Quote\Model\Quote` class)
+    - `exception` is an exception object (`\Exception` class)
+
+- `prepare_catalog_product_collection_prices` event in the `\Magento\Quote\Model\ResourceModel\Quote\Item\Collection::_assignProducts` method. Parameters:
+    - `collection` is a product collection object (`\Magento\Quote\Model\ResourceModel\Quote\Item\Collection` class)
+    - `store_id` is a store ID (`int` type)
+
+- `sales_quote_item_collection_products_after_load` event in the `\Magento\Quote\Model\QuoteManagement::_assignProducts` method. Parameters:
+    - `collection` is a product collection object (`\Magento\Catalog\Model\ResourceModel\Product\Collection` class)
+
+For information about an event in Magento 2, see [Events and observers](http://devdocs.magento.com/guides/v2.4/extension-dev-guide/events-and-observers.html#events).   
+
+### Public APIs
+
+#### Data
+
+- `\Magento\Quote\Api\Data\AddressAdditionalDataInterface`
+    - provides additional data with quote address information
+
+- `\Magento\Quote\Api\Data\AddressInterface`
+    - quote address data
+
+- `\Magento\Quote\Api\Data\CartInterface`
+    - quote data
+
+- `\Magento\Quote\Api\Data\CartItemInterface`
+    - quote item data
+  
+- `\Magento\Quote\Api\Data\CartSearchResultsInterfac`
+    - quote search result data
+
+- `\Magento\Quote\Api\Data\CurrencyInterface`
+    - currency data
+  
+- `\Magento\Quote\Api\Data\EstimateAddressInterface`
+    - estimate address data
+  
+- `\Magento\Quote\Api\Data\PaymentInterface`
+    - payment data
+  
+- `\Magento\Quote\Api\Data\PaymentMethodInterface`
+    - payment method data
+  
+- `\Magento\Quote\Api\Data\ProductOptionInterface`
+    - product option data
+  
+- `\Magento\Quote\Api\Data\ShippingAssignmentInterface`
+    - shipping assigment data
+
+- `\Magento\Quote\Api\Data\ShippingInterface`
+    - shipping data
+
+- `\Magento\Quote\Api\Data\ShippingMethodInterface`
+    - shipping method data
+  
+- `\Magento\Quote\Api\Data\TotalsAdditionalDataInterface`
+    - provides additional data for totals collection
+
+- `\Magento\Quote\Api\Data\TotalSegmentInterface`
+    - total segment data
+  
+- `\Magento\Quote\Api\Data\TotalsInterfacee`
+    - quote totals data
+  
+- `\Magento\Quote\Api\Data\TotalsItemInterface`
+    - quote items totals data
+
+#### General
+
+- `\Magento\Quote\Api\ChangeQuoteControlInterface`
+    - checks if user is allowed to change the quote
+    
+#### Guest
+
+- `\Magento\Quote\Api\GuestBillingAddressManagementInterfacee`
```

Comment:
a typo - Interfacee instead of Interface (not that it matters that much, but still)